Tests/SpeziSchedulerTests/SchedulerTests.swift

```diff
@@ -286,18 +286,49 @@ final class SchedulerTests: XCTestCase {
         let taskId = UUID()
         let scheduledDate = Date()
         let scheduledEvent = Event(taskId: taskId, scheduledAt: scheduledDate)
-        
+
         let completedEvent = Event(taskId: taskId, scheduledAt: scheduledDate)
         completedEvent.complete(true)
-        
+
         var scheduledEventHasher = Hasher()
         scheduledEvent.hash(into: &scheduledEventHasher)
         let scheduledEventHash = scheduledEventHasher.finalize()
-        
+
         var completedEventHasher = Hasher()
         completedEvent.hash(into: &completedEventHasher)
         let completedEventHash = completedEventHasher.finalize()
-        
+
         XCTAssertEqual(scheduledEventHash, completedEventHash)
     }
+    func testTaskSchedulingWithStartDateAndEndDate() async throws {
+        let startDate = Date().addingTimeInterval(3_000)
```

Comment:
A general suggestion for writing the tests: I would avoid using `.now` (or `Date()`) for dates. We had some issues with this here and in other packages. I would rather load dates based on fixed date and time combinations and use that for reasoning in the tests. Makes testing and communicating about the tests + asserting correct responses way easier.